src/components/include/DefaultImportModalDialog.jsx

```diff
@@ -0,0 +1,323 @@
+            <Dialog
+                open={show}
+                transitionDuration={0}
+                PaperProps={{ style: paperStyle }}
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {..._.omit(this, "title", "show", "onClose", "onSubmit")}
+            >
+                <DialogTitle>{this.title || title}</DialogTitle>
+
+                <DialogContent>
+                    <FormControl variant="standard" sx={{ width: "100%", alignContent: "center" }}>
+                        {fileNames.length > 0 ? (
+                            <div
+                                onDragOver={this.handleDragOver}
+                                onDragLeave={this.handleDragLeave}
+                                onDrop={this.handleDrop}
+                            >
+                                <DataGrid
+                                    rows={rows}
+                                    columns={columns}
+                                    pageSize={1}
+                                    style={dataGridStyle}
+                                />
+                            </div>
+                        ) : (
+                            <div>
+                                <label htmlFor="fileapi">
+                                    <div
+                                        onDragOver={this.handleDragOver}
+                                        onDragLeave={this.handleDragLeave}
+                                        onDrop={this.handleDrop}
+                                        style={dropZoneStyle}
+                                    >
+                                        Drop files here or click to upload
+                                    </div>
+                                </label>
+                                <input
+                                    style={{ display: "none" }}
+                                    type="file"
+                                    id="fileapi"
+                                    hidden
+                                    multiple
+                                    onChange={(event) => this.handleFileChange(event.target.files)}
+                                />
+                            </div>
+                        )}
+                    </FormControl>
+                </DialogContent>
+
+                <DialogActions>
+                    <Button data-name="Cancel" onClick={onClose}>
+                        Cancel
+                    </Button>
+                    <Button data-name="Submit" onClick={this.onSubmit || onSubmit}>
+                        Ok
+                    </Button>
+                </DialogActions>
+            </Dialog>
+        );
+    }
+}
+
+DefaultImportModalDialog.propTypes = {
+    title: PropTypes.string.isRequired,
+    show: PropTypes.bool.isRequired,
+    onClose: PropTypes.func.isRequired,
+    onSubmit: PropTypes.func.isRequired,
+    // material: PropTypes.object.isRequired,
```

Comment:
we try to use in new components with TS just interfaces for props validation, adjust this please.